tests/fixture/tmpnet/network.go

```diff
@@ -249,117 +178,84 @@ func (n *Network) PopulateNetworkConfig(networkID uint32, nodeCount int, keyCoun
+	// Configure the networking for each node and start
+	for _, node := range n.Nodes {
+		if err := n.StartNode(ctx, w, node); err != nil {
+			return err
 		}
 	}
 
-	// Ensure the node's data dir is configured
-	dataDir := node.getDataDir()
-	if len(dataDir) == 0 {
-		// NodeID will have been set by EnsureKeys
-		dataDir = filepath.Join(nodeParentDir, node.NodeID.String())
-		flags[config.DataDirKey] = dataDir
+	if _, err := fmt.Fprintf(w, "Waiting for all nodes to report healthy...\n\n"); err != nil {
+		return err
+	}
+	if err := n.WaitForHealthy(ctx, w); err != nil {
+		return err
+	}
+	if _, err := fmt.Fprintf(w, "\nStarted network %d @ %s\n", n.Genesis.NetworkID, n.Dir); err != nil {
+		return err
 	}
 
 	return nil
 }
 
-// Starts a network for the first time
-func (n *Network) Start(w io.Writer) error {
-	if len(n.Dir) == 0 {
-		return errNetworkDirNotSet
+func (n *Network) AddEphemeralNode(ctx context.Context, w io.Writer, flags FlagsMap) (*Node, error) {
+	node := NewNode("")
+	node.Flags = flags
+	node.IsEphemeral = true
+	if err := n.StartNode(ctx, w, node); err != nil {
+		return nil, err
 	}
+	return node, nil
+}
 
-	// Ensure configuration on disk is current
-	if err := n.WriteAll(); err != nil {
+// Starts the provided node after configuring it for the network.
+func (n *Network) StartNode(ctx context.Context, w io.Writer, node *Node) error {
+	if err := n.EnsureNodeConfig(node); err != nil {
 		return err
 	}
 
-	// Accumulate bootstrap nodes such that each subsequently started
-	// node bootstraps from the nodes previously started.
-	//
-	// e.g.
-	// 1st node: no bootstrap nodes
-	// 2nd node: 1st node
-	// 3rd node: 1st and 2nd nodes
-	// ...
-	//
-	bootstrapIDs := make([]string, 0, len(n.Nodes))
-	bootstrapIPs := make([]string, 0, len(n.Nodes))
-
-	// Configure networking and start each node
-	for _, node := range n.Nodes {
-		// Update network configuration
-		node.SetNetworkingConfig(bootstrapIDs, bootstrapIPs)
+	bootstrapIPs, bootstrapIDs, err := n.getBootstrapIPsAndIDs(node)
+	if err != nil {
+		return err
+	}
+	node.SetNetworkingConfig(bootstrapIDs, bootstrapIPs)
 
-		// Write configuration to disk in preparation for node start
-		if err := node.Write(); err != nil {
-			return err
-		}
+	if err := node.Write(); err != nil {
+		return err
+	}
 
-		// Start waits for the process context to be written which
-		// indicates that the node will be accepting connections on
-		// its staking port. The network will start faster with this
-		// synchronization due to the avoidance of exponential backoff
-		// if a node tries to connect to a beacon that is not ready.
-		if err := node.Start(w); err != nil {
-			return err
+	if err := node.Start(w); err != nil {
+		// Attempt to stop an unhealthy node to provide some assurance to the caller
+		// that an error condition will not result in a lingering process.
+		stopErr := node.Stop(ctx)
+		if stopErr != nil {
+			err = errors.Join(err, stopErr)
```

Comment:
I don't think we need this `stopErr != nil` check. The spec says: "Join returns an error that wraps the given errors. Any nil error values are discarded."